### ISSUE TITLE
readJpegSegments throws java.lang.NegativeArraySizeException

### ISSUE THREAD

**Issue by chenlingmin** [drewnoakes/metadata-extractor#242]
```
Exception in thread "main" java.lang.NegativeArraySizeException
	at com.drew.metadata.jpeg.JpegDhtReader.getBytes(JpegDhtReader.java:88)
	at com.drew.metadata.jpeg.JpegDhtReader.extract(JpegDhtReader.java:77)
	at com.drew.metadata.jpeg.JpegDhtReader.readJpegSegments(JpegDhtReader.java:50)
	at com.drew.imaging.jpeg.JpegMetadataReader.processJpegSegmentData(JpegMetadataReader.java:134)
	at com.drew.imaging.jpeg.JpegMetadataReader.process(JpegMetadataReader.java:126)
	at com.drew.imaging.jpeg.JpegMetadataReader.readMetadata(JpegMetadataReader.java:77)
	at com.drew.imaging.jpeg.JpegMetadataReader.readMetadata(JpegMetadataReader.java:84)
	at com.drew.imaging.ImageMetadataReader.readMetadata(ImageMetadataReader.java:111)
	at com.drew.imaging.ImageMetadataReader.readMetadata(ImageMetadataReader.java:89)
```

**Comment by drewnoakes** [comment]
Can you attach an image that reproduces this issue?

@Nadahar does this error mean anything to you?

**Comment by Nadahar** [comment]
@drewnoakes It seems like another [signed-byte bug](https://github.com/drewnoakes/metadata-extractor/blob/master/Source/com/drew/metadata/jpeg/JpegDhtReader.java#L75) by me. It should be easy to fix, but a test image would be nice to have to verify.

**Comment by chenlingmin** [comment]
![Uploading 738060981191472375.jpg…]()
This is a photo of compressed.

**Comment by chenlingmin** [comment]
![738060981191472375](https://cloud.githubusercontent.com/assets/8233283/22927412/14b497d2-f2ec-11e6-80ea-605318beaa16.jpg)
This is a photo of compressed.

**Comment by drewnoakes** [comment]
Fixed in 2.10.1 thanks to @Nadahar in #243.

**Comment by drewnoakes** [comment]
@chenlingmin thanks for the bug report by the way :)
